openpype/hosts/maya/api/lib.py

```diff
@@ -2366,6 +2366,29 @@ def set_context_settings():
     set_colorspace()
 
 
+def build_first_workfile_from_template_builder():
+    project_name = get_current_project_name()
+    project_settings = get_project_settings(project_name)
+
+    asset_name = get_current_asset_name()
+    asset = get_asset_by_name(project_name, asset_name)
+    task_name = get_current_task_name()
+    current_tasks = asset.get("data").get("tasks")
+
+    build_workfile_profiles = project_settings['maya']['templated_workfile_build'].get('profiles')  # noqa
+
+    if build_workfile_profiles:
+        for profile in build_workfile_profiles:
+            is_task_name = task_name in profile['task_names']
+            is_task_type = current_tasks[task_name]['type'] in \
+                profile['task_types']
+            is_create_first_version = profile['autobuild_first_version']
+
+            if (is_task_name or is_task_type) and is_create_first_version:
+                from .workfile_template_builder import build_workfile_template
+                build_workfile_template()
```

Comment:
```suggestion
    project_name = get_current_project_name()
    project_settings = get_project_settings(project_name)
    
    build_workfile_profiles = project_settings['maya']['templated_workfile_build'].get('profiles')  # noqa
    if not build_workfile_profiles:
        return

    asset_name = get_current_asset_name()
    asset = get_asset_by_name(project_name, asset_name)
    task_name = get_current_task_name()
    current_tasks = asset.get("data").get("tasks")
    task_type = current_tasks[task_name]['type']

    for profile in build_workfile_profiles:
        if not profile['autobuild_first_version']:
            # Do not use this profile for autobuild
            continue
    
        is_task_name = task_name in profile['task_names']
        is_task_type = task_type in profile['task_types']

        if is_task_name or is_task_type:
            # import here to avoid cyclic imports
            from .workfile_template_builder import build_workfile_template
            build_workfile_template()
```
Some optimization by opting out early whenever we can and hopefully restructures it to be more readable too.

Also, shouldn't this use regular profiles filtering for consistency? Using the `filter_profiles` function from OpenPype?